src/Heap.js

```diff
@@ -1,78 +1,79 @@
 export default class Heap {
     heap = [];
-    _nodeToIndexMap = new Map();
+    _nodeIndicies = new Map();
 
-    extractRoot(){
+    get length() {
+        return this.heap.length;
+    }
+
+    extractRoot() {
         this.swap(0, this.heap.length - 1);
 
         const node = this.heap.pop();
 
-        this._nodeToIndexMap.delete(node.node);
+        this._nodeIndicies.delete(node.node);
         this.minHeapify(0);
 
         return node; 
     }
 
-    setFscore(node, fScore) {
-        const nodeIndex = this._nodeToIndexMap.get(node);
-        if (nodeIndex) {
-            this.heap[nodeIndex].f = fScore;
+    setKey(node, key) {
+        const nodeIndex = this._nodeIndicies.get(node);
+        if (nodeIndex && nodeIndex !== 0) {
```

Comment:
why do we not want to update the root?

Reply:
I don't actually remember why I guess I'll test it without

Reply:
Logically speaking it shouldn't matter and I can't remember if there was some edge case this would be impacting.